pmm/cli.py

```diff
@@ -1,38 +1,75 @@
-#-*-coding:utf-8-*-
+# -*-coding:utf-8-*-
+"""Select PyPI index server used by pip."""
 
 from __future__ import print_function
 
+import argparse
+import re
+import sys
+
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
+
 import blindspin
 import crayons
 from pick import pick
 
-from pmm import config
+from pmm.config import PipConfig
 from pmm.data import get_mirrors_data
 
 
-def format_index_url(mirror):
-    return 'https://{mirror}/simple'.format(mirror=mirror)
+def format_index_url(url):
+    if not re.match(r'\w+://', url):
+        url = 'https://' + url
+    if not url.endswith('/simple'):
+        url = url.rstrip('/') + '/simple'
+    return url
+
 
+def main(args=None):
+    ap = argparse.ArgumentParser(description=__doc__.splitlines()[0])
+    ap.add_argument(
+        '-m',
+        '--mirrors',
+        action="store_true",
+        help="download list of PyPI mirrors and add them to selection")
+    args = ap.parse_args(sys.argv[1:] if args is None else args)
 
-def main():
-    with blindspin.spinner():
-        mirrors = get_mirrors_data()
+    pip_config = PipConfig()
+    urls = OrderedDict()
+    for server in pip_config.get_index_servers():
+        f_url = format_index_url(server['index'])
+        urls[f_url] = server
 
-    current_index_url = config.get_index_url()
+    if not urls and not args.mirrors:
+        print(crayons.red("No indexes found in %s." % pip_config.path))
```

Comment:
the ending dot '.' could be removed, or it'll display sth like: `No indexes found in /Users/wong2/.pip/pip.conf.` which is a bit weird